### ISSUE TITLE
memory leak during shard install

### ISSUE THREAD

**Issue by fauno** [amberframework/micrate#46]
Hi, while trying to install micrate as a dependency for crecto, I got this message, the `shard install` took 70% RAM (out of 8GB) and was later killed by OOM.  I tracked it down to micrate, but don't have much info since `shard install -v` doesn't show anything related (I think, I'm a newbie at crystal, but I'd expected to see some messages).  Crystal version is 0.31.1

```
GC Warning: Repeated allocation of very large block (appr. size 805310464):
        May lead to memory leak and poor performance
```

**Comment by Fishdigger** [comment]
I'm also having this issue with Crystal v0.32.1.

**Comment by c-cesar** [comment]
This is a shard bug.
I had to downgrade shards from 0.9 to 0.8 to stop getting oom.

As a sidenote to get crecto working I had to fix sqlite3 version to 0.14 (posting, because shards messages about conflicting versions are horrible.
```
sqlite3:
    github: crystal-lang/crystal-sqlite3
    version: ~> 0.14.0
```

**Comment by drujensen** [comment]
could not duplicate.  closing.
